src/components/common/FullViewSnapshotButton.tsx

```diff
@@ -0,0 +1,75 @@
+import { createSnapshot, getBillingProfiles, snapshotCreateDetails } from 'actions/index';
+import { TdrState } from 'reducers';
+import TerraTooltip from 'components/common/TerraTooltip';
+import { Button } from '@mui/material';
+import React, { Dispatch } from 'react';
+import {
+  BillingProfileModel,
+  DatasetModel,
+  SnapshotRequestContentsModelModeEnum,
+} from 'generated/tdr';
+import { Action } from 'redux';
+import { connect } from 'react-redux';
+import { now } from 'lodash';
+import { useOnMount } from '../../libs/utils';
+
+interface FullViewSnapshotButtonProps {
+  dispatch: Dispatch<Action>;
+  dataset: DatasetModel;
+  billingProfiles: Array<BillingProfileModel>;
+}
+
+function FullViewSnapshotButton({
+  dataset,
+  dispatch,
+  billingProfiles,
+}: FullViewSnapshotButtonProps) {
```

Comment:
Sonar flags this as a case where the props should be marked as readonly 
```suggestion
}: Readonly<FullViewSnapshotButtonProps>) {
```

I see that this PR [adds new sonar issues](https://sonarcloud.io/project/issues?id=DataBiosphere_jade-data-repo-ui&pullRequest=1727&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true). I don't think there's a quality gate set up for this repo so we need to enforce it ourselves if we think that fixing sonar issues is important. Do we want to fix sonar issues like this?

Reply:
yes I think it's a good idea, I'll take a look through the sonar issues.